docs/pages/react/solana-wallets/signing-and-sending-solana.mdx

```diff
@@ -0,0 +1,277 @@
+    data,
+    signMessage,
+    data: signedMessage,
+    reset,
+  } = useSolanaSignMessage({});
+
+  if (!signer) {
+    return <div>Loading signer...</div>;
+  }
+
+  return (
+    <div>
+      <button onClick={() => signMessage({ message: "Here is my message" })}>
+        {" "}
+        Sign{" "}
+      </button>
+      Message Signed: {signedMessage}
+    </div>
+  );
+}
+```
+
+```tsx
+import { useSolanaSignMessage } from "@account-kit/react";
+import { SolanaSigner } from "@account-kit/signer";
+
+function MyComponent() {
+  const {
+    signer,
+    data,
+    signMessage,
+    data: signedMessage,
+    reset,
+  } = useSolanaSignMessage({});
+
+  if (!signer) {
+    return <div>Loading signer...</div>;
+  }
+
+  return (
+    <div>
+      <button onClick={() => signMessage({ message: "Here is my message" })}>
+        {" "}
+        Sign{" "}
+      </button>
+      Message Signed: {signedMessage}
+    </div>
+  );
+}
+```
+
+# Multi-signature support using PreSign
+
+You can implement multi-signature behavior using the `PreSign` feature to authorize multiple signers on a transaction before execution.
+
+### Example use case: adding additional signers
+
+A common pattern involves supporting multiple signers for a single transaction. For example, in our demo environment, we generate a temporary (or "throwaway") account that can hold various tokens. However, this account alone isn't sufficient for transaction security or coordination.
+
+To enable multi-signature functionality, we attach additional signers to the transaction using `preSign`. This allows the transaction to be authorized by multiple parties before being broadcast, ensuring that:
+
+- Multiple stakeholders can approve a transaction.
+- Tokens can exist at a shared account address with controlled access.
+- Coordination between signers happens off-chain via pre-signed approvals.
+
+This pattern is useful for collaborative workflows, DAO interactions, or shared treasury use cases.
+
+**Example flow:**
+
+1. Create a transaction from a temporary account.
+2. Add one or more `preSign` signatures from required co-signers.
+3. Submit the fully signed transaction.
+
+[Learn more about Solana multisig.](https://blog.keyst.one/a-deep-dive-into-understanding-solana-multisig)
+
+```tsx [add-solana-signer.ts]
+import { Transaction, VersionedTransaction } from "@solana/web3.js";
+/**
+ * Send a transaction with a custom `preSend` hook to add an extra signer.
+ */
+const tx = await sendTransactionAsync({
+  instructions, // Standard transaction instructions
+  transactionComponents: {
+    preSend: addStakeSignature(stakeAccount), // Add secondary signer before sending
+  },
+});
+
+/**
+ * Returns a preSend hook that appends a signature from the provided `stakeAccount`.
+ *
+ * @param stakeAccount - The Keypair used to sign the transaction
+ * @returns A function that accepts a transaction and returns a signed transaction
+ */
+function addStakeSignature(stakeAccount: Keypair) {
```

Comment:
This naming/structure is a bit confusing to me. I would expect a function named `addStakeSignature` to actually add a signature, but instead it's returning a function which can later be called to add the signature. I might be explicit and name it something like `getAddStakeSignatureFn`.

Reply:
I'm bad at naming; this seems reasonable, thanks.